TunnelKit/Sources/Protocols/OpenVPN/TLSBox.m

```diff
@@ -397,4 +410,70 @@ - (BOOL)verifyEKUWithSSL:(SSL *)ssl
     return isValid;
 }
 
+#pragma mark SAN
+
+- (BOOL)verifySANHostWithSSL:(SSL *)ssl {
+    X509 *cert = SSL_get_peer_certificate(self.ssl);
+    if (!cert) {
+        return NO;
+    }
+    
+    GENERAL_NAMES* names = NULL;
+    unsigned char* utf8 = NULL;
+    names = X509_get_ext_d2i(cert, NID_subject_alt_name, 0, 0 );
+    if(!names) {
+        X509_free(cert);
+        return NO;
+    }
+    
+    int i = 0, count = sk_GENERAL_NAME_num(names);
+    if(!count) {
+        X509_free(cert);
+        GENERAL_NAMES_free(names);
+        return NO;
+    }
+    BOOL isValid = NO;
+    
+    for( i = 0; i < count; ++i )    {
+        GENERAL_NAME* entry = sk_GENERAL_NAME_value(names, i);
+        if(!entry) {
+            continue;
+        }
+        if(GEN_DNS != entry->type) {
+            continue;
+        }
+        
+        int len1 = 0, len2 = -1;
+        len1 = ASN1_STRING_to_UTF8(&utf8, entry->d.dNSName);
+        if(!utf8) {
+            continue;
+        }
+        len2 = (int)strlen((const char*)utf8);
+        
+        if(len1 != len2) {
+            OPENSSL_free(utf8);
+            utf8 = NULL;
+            continue;
+        }
+        
+        if(utf8 && len1 && len2 && (len1 == len2) && strcmp(utf8, self.hostname.UTF8String) == 0) {
```

Comment:
Prepend `utf8` argument to `strcmp` with a `(const char *)` casting to suppress a warning.